### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by jmf373**
- Busy time histogram label no longer cut off
- Better internal histogram layout 
Fixed:
- If you select a gym → click on hours drop down → tap on a row, the row stays grey
- If you click on the classes page → click on search “find a way to sweat” → no way to dismiss the keyboard and go back to the screen
- Reset button in filter screen dismisses view
- Back/favorite button in detail view changes color to adjust for dark/light backgrounds

### CODE REVIEW

File: Fitness/Views/Histogram.swift [hunk 1]
```diff
@@ -30,13 +30,15 @@ class Histogram: UIView {
...
 
     var bottomAxis: UIView!
     var bottomAxisTicks: [UIView]!
 
     init(frame: CGRect, data: [Int], todaysHours: DailyGymHours) {
         super.init(frame: frame)
         self.data = data
+        self.hours = todaysHours
 
         //AXIS
```
Comment (young-k) [inline Fitness/Views/Histogram.swift]: shouldn't there be a space after `//`?

File: Fitness/Views/Histogram.swift [hunk 2]
```diff
@@ -82,13 +84,14 @@ class Histogram: UIView {
...
         selectedTime.font = ._12LatoBold
         selectedTime.textAlignment = .right
-        if (openHour + selectedIndex) < 12 {
-            selectedTime.text = String(openHour + selectedIndex) + "AM :"
-        } else if (openHour + selectedIndex) > 12 {
-            selectedTime.text = String(openHour + selectedIndex - 12) + "PM :"
-        } else {
-            selectedTime.text = "12PM :"
-        }
+//        if (openHour + selectedIndex) < 12 {
```
Comment (young-k) [inline Fitness/Views/Histogram.swift]: why not delete the commented out code?

File: Fitness/Views/Histogram.swift [hunk 3]
```diff
@@ -116,8 +119,8 @@ class Histogram: UIView {
     // MARK: - CONSTRAINTS
     func setupConstraints() {
         //AXIS
```
Comment (young-k) [inline Fitness/Views/Histogram.swift]: same here

File: Fitness/Views/Histogram.swift [hunk 4]
```diff
@@ -217,14 +221,8 @@ class Histogram: UIView {
...
-
-        if (openHour + selectedIndex) < 12 {
-            selectedTime.text = String(openHour + selectedIndex) + "AM :"
-        } else if (openHour + selectedIndex) > 12 {
-            selectedTime.text = String(openHour + selectedIndex - 12) + "PM :"
-        } else {
-            selectedTime.text = "12PM :"
-        }
+        
+        selectedTime.text = hours.openTime.addingTimeInterval( Double(selectedIndex) * 3600 ).getStringOfDatetime(format: "ha :")
```
Comment (young-k) [inline Fitness/Views/Histogram.swift]: shouldn't`3600` be defined in a constant at the top?

File: Fitness/Controllers/HomeController.swift [hunk 5]
```diff
@@ -286,8 +286,8 @@ extension HomeController: UICollectionViewDelegate, UICollectionViewDelegateFlow
 
 // MARK: - NavigationDelegate
 extension HomeController: NavigationDelegate {
-    func changeTab(index: Int) {
-        tabBarController?.selectedIndex = index
+    func viewTodaysClassse() {
```
Comment (AAAstorga) [inline Fitness/Controllers/HomeController.swift]: typo? should it be `viewTodaysClasses`?

File: Fitness/Views/Headers/TodaysClassesHeaderView.swift [hunk 6]
```diff
@@ -9,7 +9,7 @@
 import UIKit
 
 protocol NavigationDelegate {
-    func changeTab(index: Int)
+    func viewTodaysClassse()
```
Comment (AAAstorga) [inline Fitness/Views/Headers/TodaysClassesHeaderView.swift]: here as well

File: Fitness/Views/Histogram.swift (hunk 2 above)
Comment (AAAstorga) [reply Fitness/Views/Histogram.swift]: +1

File: Fitness/Views/Histogram.swift [hunk 7]
```diff
@@ -82,13 +84,14 @@ class Histogram: UIView {
...
-        }
+//        if (openHour + selectedIndex) < 12 {
+//            selectedTime.text = String(openHour + selectedIndex) + "AM :"
+//        } else if (openHour + selectedIndex) > 12 {
+//            selectedTime.text = String(openHour + selectedIndex - 12) + "PM :"
+//        } else {
+//            selectedTime.text = "12PM :"
+//        }
+        selectedTime.text = todaysHours.openTime.addingTimeInterval( Double(selectedIndex) * 3600 ).getStringOfDatetime(format: "ha :")
         selectedTime.sizeToFit()
```
Comment (AAAstorga) [inline Fitness/Views/Histogram.swift]: shouldn't be using `sizeToFit()` if you are using autolayout correctly

File: Fitness/Views/Histogram.swift [hunk 8]
```diff
@@ -127,8 +130,8 @@ class Histogram: UIView {
 
             let tick = bottomAxisTicks[i]
 
-            tick.snp.updateConstraints {make in
-                if (i == 0) {
+            tick.snp.makeConstraints { make in
+                if i == 0 {
                     make.left.equalToSuperview().offset(tickSpacing)
```
Comment (AAAstorga) [inline Fitness/Views/Histogram.swift]: leading 😠

Comment (AAAstorga) [reply Fitness/Views/Histogram.swift]: lol I'm not mad I just like that emoji

File: Fitness/Views/Histogram.swift [hunk 9]
```diff
@@ -167,31 +170,32 @@ class Histogram: UIView {
     func setupSelectedConstraints() {
 
         if selectedIndex < bars.count {
-            selectedLine.snp.remakeConstraints {make in
+            selectedLine.snp.remakeConstraints { make in
```
Comment (AAAstorga) [inline Fitness/Views/Histogram.swift]: remake or make?

File: Fitness/Views/Histogram.swift [hunk 10]
```diff
@@ -217,14 +221,8 @@ class Histogram: UIView {
...
-        if (openHour + selectedIndex) < 12 {
-            selectedTime.text = String(openHour + selectedIndex) + "AM :"
-        } else if (openHour + selectedIndex) > 12 {
-            selectedTime.text = String(openHour + selectedIndex - 12) + "PM :"
-        } else {
-            selectedTime.text = "12PM :"
-        }
+        
+        selectedTime.text = hours.openTime.addingTimeInterval( Double(selectedIndex) * 3600 ).getStringOfDatetime(format: "ha :")
         selectedTime.sizeToFit()
```
Comment (AAAstorga) [inline Fitness/Views/Histogram.swift]: you shouldn't need to use `sizeToFit()`

File: Fitness/Views/Histogram.swift (hunk 9 above)
Comment (jmf373) [reply Fitness/Views/Histogram.swift]: It's called anytime the user selects a bar, so it should be remake, right?

**Pull request merged**